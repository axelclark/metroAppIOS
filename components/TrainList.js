import React from 'react';
import {
  FlatList,
  StyleSheet,
  Text,
  View,
} from 'react-native';
import { COLOR_PRIMARY } from '../constants/styles'

const Arrival = ({ mins }) => {
  switch (mins) {
    case 'BRD':
      return <Text style={styles.text}>Boarding!</Text>
    case 'ARR':
      return <Text style={styles.text}>Train is arriving...</Text>
    case '1':
      return <Text style={styles.text}>Arrival in 1 minute</Text>
    case '':
      return <Text style={styles.text}>Arrival in TBD minutes</Text>
    default:
      return <Text style={styles.text}>Arrival in {mins} minutes</Text>
  }
}

const Train = ({ train }) => (
  <View style={styles.container}>
    <Text style={styles.text}>{train.Name}</Text>
    <Text style={styles.text}>Destination: {train.DestinationName}</Text>
    <Arrival mins={train.Min} />
  </View>
)

const TrainList = ({ trains }) => {
  if (trains.length) {
    const renderTrains = trains.map(train =>
      (<Train
        key={`${train.DestinationCode}-${train.Min}`}
        train={train}
      />))

    return (
      <View>
        {renderTrains}
      </View>
    )
  }

  return (
    <View style={styles.container}>
      <Text style={styles.text}>No trains are currently scheduled</Text>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
  },
  text: {
    color: COLOR_PRIMARY,
    fontSize: 18,
  },
});

export default TrainList
