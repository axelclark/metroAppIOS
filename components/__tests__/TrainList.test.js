import React from 'react'
import TrainList from '../TrainList'

import renderer from 'react-test-renderer'

test('renders correctly', () => {
  const trains = [
    {
      "Car": "8",
      "Destination": "Shady Gr",
      "DestinationCode": "A15",
      "DestinationName": "Shady Grove",
      "Group": "2",
      "Line": "RD",
      "LocationCode": "A05",
      "LocationName": "Cleveland Park",
      "Min": "BRD"
    },
    {
      "Car": "8",
      "Destination": "Shady Gr",
      "DestinationCode": "A15",
      "DestinationName": "Shady Grove",
      "Group": "2",
      "Line": "RD",
      "LocationCode": "A05",
      "LocationName": "Cleveland Park",
      "Min": "ARR"
    },
    {
      "Car": "8",
      "Destination": "Shady Gr",
      "DestinationCode": "A15",
      "DestinationName": "Shady Grove",
      "Group": "2",
      "Line": "RD",
      "LocationCode": "A09",
      "LocationName": "Bethesda",
      "Min": "10"
    },
    {
      "Car": "8",
      "Destination": "Shady Gr",
      "DestinationCode": "A15",
      "DestinationName": "Shady Grove",
      "Group": "2",
      "Line": "RD",
      "LocationCode": "A09",
      "LocationName": "Bethesda",
      "Min": ""
    }
  ]
  const tree = renderer.create(<TrainList trains={trains}/>).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders when no trains scheduled', () => {
  const trains = []
  const tree = renderer.create(<TrainList trains={trains}/>).toJSON()
  expect(tree).toMatchSnapshot()
})
