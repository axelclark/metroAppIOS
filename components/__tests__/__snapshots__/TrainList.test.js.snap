// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#171f3d",
          "fontSize": 18,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#171f3d",
          "fontSize": 18,
        }
      }
    >
      Destination: 
      Shady Grove
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#171f3d",
          "fontSize": 18,
        }
      }
    >
      Boarding!
    </Text>
  </View>
  <View
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#171f3d",
          "fontSize": 18,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#171f3d",
          "fontSize": 18,
        }
      }
    >
      Destination: 
      Shady Grove
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#171f3d",
          "fontSize": 18,
        }
      }
    >
      Train is arriving...
    </Text>
  </View>
  <View
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#171f3d",
          "fontSize": 18,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#171f3d",
          "fontSize": 18,
        }
      }
    >
      Destination: 
      Shady Grove
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#171f3d",
          "fontSize": 18,
        }
      }
    >
      Arrival in 
      10
       minutes
    </Text>
  </View>
  <View
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#171f3d",
          "fontSize": 18,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#171f3d",
          "fontSize": 18,
        }
      }
    >
      Destination: 
      Shady Grove
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#171f3d",
          "fontSize": 18,
        }
      }
    >
      Arrival in TBD minutes
    </Text>
  </View>
</View>
`;

exports[`renders when no trains scheduled 1`] = `
<View
  style={Object {}}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#171f3d",
        "fontSize": 18,
      }
    }
  >
    No trains are currently scheduled
  </Text>
</View>
`;
